nestjs/src/cross-check/cross-check.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Repository } from 'typeorm';
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { HttpService } from '@nestjs/axios';
+import { Cron } from '@nestjs/schedule';
+import { ConfigService } from '../config';
+import { isTaskNeededToStart, isTaskNeededToFinish } from './utils/tasks-filtering';
+import { CourseTask } from '@entities/courseTask';
+import { from, catchError, mergeMap, EMPTY, toArray, lastValueFrom } from 'rxjs';
+
+@Injectable()
+export class CrossCheckService {
+  private readonly logger = new Logger('CrossCheckService');
+
+  constructor(
+    @InjectRepository(CourseTask)
+    private readonly courseTaskRepository: Repository<CourseTask>,
+    private readonly httpService: HttpService,
+    private readonly conf: ConfigService,
+  ) {}
+
+  @Cron('5 0 * * *', { timeZone: 'UTC' })
+  async executeCronJobs() {
+    const { tasksToStart, tasksToFinish } = await this.getCrossCheckTasks();
+    await this.initCrossCheckAction(tasksToStart, 'distribution');
+    await this.initCrossCheckAction(tasksToFinish, 'completion');
+  }
+
+  private getInitalDataForRequest() {
+    const { username, password } = this.conf.users.root;
+
+    const authHeader = `Basic ${Buffer.from(username + ':' + password).toString('base64')}`;
```

Comment:
Please use built-in axios functionality instead of generating  manually. See `https://github.com/axios/axios`

```
// `auth` indicates that HTTP Basic auth should be used, and supplies credentials.
  // This will set an `Authorization` header, overwriting any existing
  // `Authorization` custom headers you have set using `headers`.
  // Please note that only HTTP Basic auth is configurable through this parameter.
  // For Bearer tokens and such, use `Authorization` custom headers instead.
  auth: {
    username: 'janedoe',
    password: 's00pers3cret'
  },
```